test/extensions/load_balancing_policies/client_side_weighted_round_robin/client_side_weighted_round_robin_lb_test.cc

```diff
@@ -0,0 +1,2047 @@
+        std::make_shared<ClientSideWeightedRoundRobinLoadBalancer::WorkerLocalLb>(
+            priority_set_, local_priority_set_.get(), stats_, runtime_, random_, common_config_,
+            lb_config_, simTime()));
+
+    // Initialize the thread aware load balancer from config.
+    ASSERT_EQ(lb_->initialize(), absl::OkStatus());
+  }
+
+  // Updates priority 0 with the given hosts and hosts_per_locality.
+  void updateHosts(HostVectorConstSharedPtr hosts,
+                   HostsPerLocalityConstSharedPtr hosts_per_locality) {
+    local_priority_set_->updateHosts(
+        0,
+        updateHostsParams(hosts, hosts_per_locality,
+                          std::make_shared<const HealthyHostVector>(*hosts), hosts_per_locality),
+        {}, empty_host_vector_, empty_host_vector_, random_.random(), absl::nullopt);
+  }
+
+  void peekThenPick(std::vector<int> picks) {
+    for (auto i : picks) {
+      EXPECT_EQ(hostSet().healthy_hosts_[i], lb_->peekAnotherHost(nullptr));
+    }
+    for (auto i : picks) {
+      EXPECT_EQ(hostSet().healthy_hosts_[i], lb_->chooseHost(nullptr));
+    }
+  }
+
+  envoy::extensions::load_balancing_policies::client_side_weighted_round_robin::v3::
+      ClientSideWeightedRoundRobin client_side_weighted_round_robin_config_;
+
+  std::shared_ptr<PrioritySetImpl> local_priority_set_;
+  std::shared_ptr<ClientSideWeightedRoundRobinLoadBalancerFriend> lb_;
+  HostsPerLocalityConstSharedPtr empty_locality_;
+  HostVector empty_host_vector_;
+
+  NiceMock<MockLoadBalancerContext> lb_context_;
+  NiceMock<Event::MockDispatcher> dispatcher_;
+  NiceMock<MockClusterInfo> cluster_info_;
+  ClientSideWeightedRoundRobinLbConfig lb_config_ =
+      ClientSideWeightedRoundRobinLbConfig(client_side_weighted_round_robin_config_, dispatcher_);
+};
+
+//////////////////////////////////////////////////////
+// These tests are copied from ../round_robin/round_robin_lb_test.cc to test common functionality.
+//
+// For the tests which mutate primary and failover host sets explicitly, only
+// run once.
+using FailoverTest = ClientSideWeightedRoundRobinLoadBalancerTest;
+
+// Ensure if all the hosts with priority 0 unhealthy, the next priority hosts
+// are used.
+TEST_P(FailoverTest, BasicFailover) {
+  host_set_.hosts_ = {makeTestHost(info_, "tcp://127.0.0.1:80", simTime())};
+  failover_host_set_.healthy_hosts_ = {makeTestHost(info_, "tcp://127.0.0.1:82", simTime())};
+  failover_host_set_.hosts_ = failover_host_set_.healthy_hosts_;
+  init(false);
+  // DeterminePriorityLoad is called in chooseHost.
+  HealthyAndDegradedLoad priority_load{Upstream::HealthyLoad({0, 100}),
+                                       Upstream::DegradedLoad({0, 0})};
+  ON_CALL(lb_context_, determinePriorityLoad(_, _, _)).WillByDefault(ReturnRef(priority_load));
+
+  EXPECT_EQ(failover_host_set_.healthy_hosts_[0], lb_->peekAnotherHost(&lb_context_));
+  EXPECT_EQ(failover_host_set_.healthy_hosts_[0], lb_->chooseHost(&lb_context_));
+}
```

Comment:
nit: Basically, the new lb inherit from round robin, so, I think we only need to copy few core tests here.